src/daft-logical-plan/src/optimization/optimizer.rs

```diff
@@ -100,6 +101,7 @@ impl Default for OptimizerBuilder {
                         Box::new(EliminateSubqueryAliasRule::new()),
                         Box::new(SplitActorPoolProjects::new()),
                         Box::new(DetectMonotonicId::new()),
+                        Box::new(ExtractWindowFunction::new()),
```

Comment:
For all three of these rules, it would actually make the most sense to put them at the bottom, after all of the other rules (plus maybe some rule to clean up extranneous projects, but don't worry about that for this PR)